samples/app-anonymous-users/csharp/ClientApp/src/components/shareview.jsx

```diff
@@ -84,11 +109,202 @@ const ShareView = () => {
         }
     }
 
+    // Initiate facebook login.
+    const facebookLogin = () => {
+        fbAuthentication() // This method get a client-side token for Facebook
+            .then((result) => {
+                return getFacebookProfileName(result.idToken); // This method get a face book user profile details.
+            })
+            .catch((error) => {
+                console.log(error);
+            });
+    }
+
+    // Get client side token for facebook.
+    const fbAuthentication = () => {
+        var facebookAppId = process.env.REACT_APP_FACEBOOK_APP_ID;
+        let redirectUri = window.location.origin + "/facebook-auth-end";
+
+        return new Promise((resolve, reject) => {
+            microsoftTeams.authentication.authenticate({
+                url: `https://www.facebook.com/v12.0/dialog/oauth?client_id=${facebookAppId}&redirect_uri=${redirectUri}&state=state`,
```

Comment:
You're still using a static value for the state here. 

There is more information on what we should be doing in this article https://auth0.com/docs/secure/attack-protection/state-parameters.

But essentially, we should

1. create a random string
2. save that string locally (in local storage, cookies or similar)
3. send the string as the state param in this oauth request
4. in facebook-auth-end, when we are reading the url paramters passed by facebook, we should compare the state they pass to the string we've stored locally.
5. If they we can trust the token, if they don't match we should throw an error

Reply:
Fixed